src/components/Auth.vue

```diff
@@ -153,10 +153,11 @@ export default {
       const headerList = { headers: {
         'Authorization': 'Bearer ' + this.accessInfo.access_token
       } };
+      const currentMonth = new Date().getMonth();
       return axios.get(process.env.FIREROAD_URL + '/verify/', headerList)
         .then(function (verifyResponse) {
           if (verifyResponse.data.success) {
-            this.$emit('set-sem', verifyResponse.data.current_semester);
+            this.$emit('set-sem', verifyResponse.data.current_semester - (currentMonth === 4 ? 1 : 0));
```

Comment:
This same logic should also apply to line 120, `this.$emit('set-sem', this.accessInfo.current_semester);`

Reply:
Good catch, thank you. I forgot the cookie was directly storing what FireRoad gave us

Reply:
Actually it turns out that line was unnecessary because `verify` is called right below it and that's where this happens.